### PR TITLE
Add Apps K8s sub-section.

### PR DESCRIPTION
**Pull request by johnharris85**
Initial PR to add some of the internal CNR content to TDC.

Signed-off-by: John Harris <joharris@vmware.com>

### CODE REVIEW

**Comment by zmb3** [comment]
Rather than put new shortcodes/styles/etc in the theme, shouldn't we override them in the top level, as per [hugo overrides](https://gohugobrasil.netlify.app/themes/customizing/#override-archetypes)?

This would mean, for example, that the aside.yml should go in `./layouts/shortcodes/` instead of `./themes/docsy/layouts/shortcodes`.

**Comment by bryanfriedman** [comment]
> Rather than put new shortcodes/styles/etc in the theme, shouldn't we override them in the top level, as per [hugo overrides](https://gohugobrasil.netlify.app/themes/customizing/#override-archetypes)?
> 
> This would mean, for example, that the aside.yml should go in `./layouts/shortcodes/` instead of `./themes/docsy/layouts/shortcodes`.
> 
> (I recognize that it's already been done this way here as well as in previous versions of the library, but it does make updating to newer versions of a theme more challenging than it should be.)

This would be right except for the fact that I think we've gone so far outside the realm of the original theme that it's not even really that theme anymore...

File: themes/docsy/assets/scss/_custom.scss [hunk 1]
```diff
@@ -903,6 +903,69 @@ body.tv-episode {
   margin-right: 10px;
 }
 
+.checklist-item {
+  padding: 2rem 2rem 1rem;
+  margin: 1.5rem 0;
```
Comment (bryanfriedman) [inline themes/docsy/assets/scss/_custom.scss:908]: To my eye this feels like too much whitespace, but maybe we can get a real designer like Rachel or Alex to take a look...

File: themes/docsy/layouts/learningpaths/baseof.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<!doctype html>
```
Comment (bryanfriedman) [inline themes/docsy/layouts/learningpaths/baseof.html:1]: Were you experimenting with a learningpaths layout? Looks like there's no content section yet so maybe we shouldn't include these files in this PR just yet?

File: themes/docsy/assets/scss/_custom.scss [hunk 3]
```diff
@@ -903,6 +903,69 @@ body.tv-episode {
...
+.checklist-item > input[type="checkbox"] {
+  margin-right: 1em;
+  margin-top: 5px;
+}
+
+.checklist-item h4 {
+  margin: 0;
+}
+
+table {
```
Comment (bryanfriedman) [inline themes/docsy/assets/scss/_custom.scss:922]: We can address this later but we might want to be careful about setting global table styling. There is a `table` class I had been using and wrapping Markdown tables with `{{< table "table" >}}` as in https://github.com/vmware-tanzu/tanzu-dev-portal/blob/main/content/guides/microservices/distributed-tracing.md

File: themes/docsy/assets/scss/_custom.scss [hunk 4]
```diff
@@ -903,6 +903,69 @@ body.tv-episode {
...
+
+div.aside div.aside-title i {
+    margin-right: 5px;
+}
+
+div.aside div.aside-content {
+    padding: 5px 20px;
+    background: #0f171c;
+    /* background-color: black; */
+}
```
Comment (bryanfriedman) [inline themes/docsy/assets/scss/_custom.scss:967]: We will need to make sure to add light mode versions of these styles to the `light-theme.css` file.

File: content/guides/kubernetes/app-enhancements-probing-app-state.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+}
+
+func start() error {
+	// app logic for starting
+	return nil
+}
+
+func isHealthy() {
+	if
```
Comment (BrianMMcClain) [inline content/guides/kubernetes/app-enhancements-probing-app-state.md:63]: Incomplete example?

File: content/guides/kubernetes/app-enhancements-spring-k8s.md [hunk 6]
```diff
@@ -0,0 +1,1151 @@
...
+
+![Microservices Landscape](/images/guides/kubernetes/app-enhancements/spring-boot-micro-services-landscape.png)
+
+## Reference Architecture Environment
+
+Each microservice runs in its own container, one container per pod and one pod
+per service replica. The application is built using a microservices architecture and
+represented by replicated containers calling each other.
+
+![Microservices Stack](/images/guides/kubernetes/app-enhancements/spring-boot-micro-services-stack.png)
```
Comment (bryanfriedman) [inline content/guides/kubernetes/app-enhancements-spring-k8s.md:94]: Can we crop these images to there isn't the extra whitespace around the border? This isn't the only image but it's one example.

File: content/guides/kubernetes/app-enhancements-probing-app-state.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+Note that `initialDelaySeconds` can be removed from the 
+`livenessProbe` as we're using `startupProbe` to cover our
+Pod initialization checks. In the example above, the `startupProbe`
+is allowing the container 300 seconds (`periodSeconds` * `failureThreshold`)
+to finish its startup. It is also validating against a different
+endpoint than the `livenessProbe`.
+
+![startup probe](/images/guides/kubernetes/app-enhancements/startup-probe.png)
```
Comment (bryanfriedman) [inline content/guides/kubernetes/app-enhancements-probing-app-state.md:195]: For images with transparent backgrounds (or even ones that need some padding) you can add them to a subfolder under `diagrams` and explained in the README (https://github.com/vmware-tanzu/tanzu-dev-portal#images). This makes it work okay in both light and dark mode.

File: content/guides/kubernetes/app-enhancements-spring-k8s.md [hunk 8]
```diff
@@ -0,0 +1,1151 @@
...
+metadata:
+  name: department
+data:
+  spring.cloud.kubernetes.discovery.all-namespaces: "true"
+  # other config removed for brevity.
+```
+
+Feign is a declarative web service client. In order to communicate with
+`employee-service` from `department-service` we need to create an interface and
+use the @FeignClient annotation. In the `@FeignClient` annotation the String
```
Comment (BrianMMcClain) [inline content/guides/kubernetes/app-enhancements-spring-k8s.md:218]:
```suggestion
use the `@FeignClient` annotation. In the `@FeignClient` annotation the String
```

**Review by BrianMMcClain — CHANGES_REQUESTED** [review]
Only had a couple things to add to what @bryanfriedman mentioned

File: themes/docsy/layouts/learningpaths/baseof.html (hunk 2 above)
Comment (johnharris85) [reply themes/docsy/layouts/learningpaths/baseof.html:1]: Gah must have committed too much, I excluded most of it. Will fix.

File: themes/docsy/assets/scss/_custom.scss (hunk 1 above)
Comment (johnharris85) [reply themes/docsy/assets/scss/_custom.scss:908]: Have changed these, they look a little tighter now.

File: themes/docsy/assets/scss/_custom.scss (hunk 3 above)
Comment (johnharris85) [reply themes/docsy/assets/scss/_custom.scss:922]: Oops missed this when searching, have altered to use your style and removed mine.

File: content/guides/kubernetes/app-enhancements-probing-app-state.md (hunk 5 above)
Comment (johnharris85) [reply content/guides/kubernetes/app-enhancements-probing-app-state.md:63]: Haha, weirdly this is like this internally as well. Guess we got distracted :joy:. Gonna remove it for now.

**Comment by johnharris85** [comment]
OK addressed all comments. For the styling ones I used best judgement (which is probably awful) so might need some touching up later from UX team.

**Comment by johnharris85** [comment]
This good to go now @bryanfriedman / @BrianMMcClain ?

**Review by BrianMMcClain — APPROVED** [review]

**Pull request merged**